pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -161,10 +164,38 @@ func getK8sStorageClassNameWithImmediateBindingModeForPolicy(ctx context.Context
 			"Either storagepolicyId: %s doesn't match any storage class, or the policy is not assigned to namespace: %s",
 			storagePolicyID, namespace)
 	} else {
-		// TODO: Implement Quota check using new Storage Quota CR before returning storage class
-		return scName, nil
+		storagePolicyQuotaList := &cnsstoragepolicyquotasv1alpha1.StoragePolicyQuotaList{}
+		err := client.List(ctx, storagePolicyQuotaList, &ctrlruntime.ListOptions{
+			Namespace: namespace,
+		})
+		if err != nil {
+			return "", logger.LogNewErrorf(log, "Failed to list StoragePolicyQuota CR on the namespace: %s", namespace)
+		}
+		foundMatchStoragePolicyQuotaCR := false
+		matchedStoragePolicyQuotaCR := cnsstoragepolicyquotasv1alpha1.StoragePolicyQuota{}
+		if scName != "" && len(storagePolicyQuotaList.Items) > 0 {
+			for _, storagePolicyQuota := range storagePolicyQuotaList.Items {
+				if storagePolicyQuota.Spec.StoragePolicyId == storagePolicyID {
+					log.Debugf("Found storagePlicyQuota CR with matching storagePolicyId:%s in namespaces:%s", storagePolicyID, namespace)
+					foundMatchStoragePolicyQuotaCR = true
+					matchedStoragePolicyQuotaCR = storagePolicyQuota
+					break
+				}
+			}
+			if foundMatchStoragePolicyQuotaCR && len(matchedStoragePolicyQuotaCR.Status.SCLevelQuotaStatuses) > 0 {
+				for _, quota := range matchedStoragePolicyQuotaCR.Status.SCLevelQuotaStatuses {
```

Comment:
can we add NOTE here to say that this code expects StoragePolicyQuota CR status will be populated with all fields with zero values from the start (to be taken care by quota controller), otherwise this code will always return error.